### PR TITLE
Np 733 add peer reviewed to publication instance

### PR DESCRIPTION
**Pull request by brinxmat**

### CODE REVIEW

File: src/main/java/no/unit/nva/model/PublicationType.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 public enum PublicationType {
 
-    JOURNAL_ARTICLE("JournalArticle");
+    JOURNAL_ARTICLE("Article");
```
Comment (janbj) [inline src/main/java/no/unit/nva/model/PublicationType.java:11]: Why is this just Article now?

File: src/test/java/no/unit/nva/model/IssnValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class IssnValidatorTest {
+
+    @DisplayName("The valid ISSNs pass")
+    @Test
+    void issnCreatesIssnWithValidInput() {
+        List<String> validIssns = List.of("1476-4687", "0363-6941", "1945-662X", "0025-5629");
```
Comment (janbj) [inline src/test/java/no/unit/nva/model/IssnValidatorTest.java:18]: You can use ParameterizedTest: https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests

File: src/main/java/no/unit/nva/model/Journal.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+import java.util.Objects;
+
+import static java.util.Objects.nonNull;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+public class Journal extends PublicationContext {
+    private String printIssn;
+    private String onlineIssn;
+
+    /* default */ Journal() {
```
Comment (axthosarouris) [inline src/main/java/no/unit/nva/model/Journal.java:16]: why not public

File: src/main/java/no/unit/nva/model/PublicationType.java (hunk 1 above)
Comment (brinxmat) [reply src/main/java/no/unit/nva/model/PublicationType.java:11]: Heavy handed refactoring, methinks, reverting.

Comment (axthosarouris) [reply src/main/java/no/unit/nva/model/PublicationType.java:11]: True. Maybe in the future we will support proceedings from conferences as well. NTNU is using conference proceedings as well

File: src/main/java/no/unit/nva/model/instancetypes/Article.java [hunk 4]
```diff
@@ -1,24 +1,27 @@
...
+package no.unit.nva.model.instancetypes;
 
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import no.unit.nva.model.pages.Range;
 
 import java.util.Objects;
 
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
-public class PublicationInstance {
+public class Article extends PublicationInstance {
```
Comment (axthosarouris) [inline src/main/java/no/unit/nva/model/instancetypes/Article.java:9]: JournalArticle maybe?

File: src/main/java/no/unit/nva/model/validator/IssnValidator.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        }
+        return testStructure(issn);
+    }
+
+    private static boolean testStructure(String issn) {
+        String value = issn.toUpperCase(Locale.ENGLISH);
+        char[] characters = value.toCharArray();
+        List<Integer> integers = new ArrayList<>();
+        for (int counter = 0; counter < characters.length - 1; counter++) {
+            if (counter == ISSN_HYPHEN_POSITION && characters[counter] != HYPHEN) {
```
Comment (axthosarouris) [inline src/main/java/no/unit/nva/model/validator/IssnValidator.java:33]: This check can be outside the loop.
```
if ( characters[ISSN_HYPHEN_POSITION]!= HYPHEN){
      return false
}

```

File: src/main/java/no/unit/nva/model/validator/IssnValidator.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            if (counter != ISSN_HYPHEN_POSITION) {
+                try {
+                    number = Integer.parseInt(String.valueOf(characters[counter]));
+                } catch (NumberFormatException e) {
+                    return false;
+                }
+                integers.add(number);
+            }
+        }
+        return generateCheckbit(integers) == characters[characters.length - 1];
```
Comment (axthosarouris) [inline src/main/java/no/unit/nva/model/validator/IssnValidator.java:46]: refactor to a method with a meaningful name

File: src/test/java/no/unit/nva/model/IssnValidatorTest.java (hunk 2 above)
Comment (axthosarouris) [reply src/test/java/no/unit/nva/model/IssnValidatorTest.java:18]: Experimental feature?

File: src/main/java/no/unit/nva/model/validator/IssnValidator.java (hunk 6 above)
Comment (brinxmat) [reply src/main/java/no/unit/nva/model/validator/IssnValidator.java:48]: I wonder what the name for generation of a checkbit would be? :D

Comment (janbj) [reply src/main/java/no/unit/nva/model/validator/IssnValidator.java:48]: Maybe he means the statement itself.

File: src/main/java/no/unit/nva/model/pages/Composite.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
+package no.unit.nva.model.pages;
+
+import java.util.Objects;
+
+public class Composite implements Pages {
```
Comment (axthosarouris) [inline src/main/java/no/unit/nva/model/pages/Composite.java:5]: Composite is a very generic name

File: src/main/java/no/unit/nva/model/instancetypes/PublicationInstance.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+@JsonSubTypes({@JsonSubTypes.Type(name = "JournalArticle", value = JournalArticle.class)})
+public class PublicationInstance {
+    private Pages pages;
+    private boolean peerReviewed;
+
+    /* default */ PublicationInstance() {
+
+    }
+
+    public Pages getPages() {
```
Comment (axthosarouris) [inline src/main/java/no/unit/nva/model/instancetypes/PublicationInstance.java:17]: Can we have a test that pages gets serialized/deserialized correctly?

File: src/main/java/no/unit/nva/model/pages/MonographPages.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    private String pages;
+    private boolean illustrated;
+
+    private MonographPages(Builder builder) {
+        setIntroduction(builder.introduction);
+        setPages(builder.pages);
+        setIllustrated(builder.illustrated);
+    }
+
+    public Range getIntroduction() {
```
Comment (axthosarouris) [inline src/main/java/no/unit/nva/model/pages/MonographPages.java:16]: add @JacocoGenerated to setters and getters?

**Pull request merged**